src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/MiningControllerTest.cs

```diff
@@ -70,31 +85,100 @@ public void Generate_Blocks_When_Model_Is_Invalid_ReturnsBadRequest()
         }
 
         [Fact]
-        public void Generate_Blocks_ReturnsSuccess()
+        public void StartMining_PowNetwork_ReturnsSuccess()
         {
-            const string wallet = "wallet";
-            const string account = "account";
+            var powNode = new Mock<IFullNode>();
 
-            this.walletManager.Setup(w => w.GetWalletsNames()).Returns(new List<string> { wallet });
-            this.walletManager.Setup(w => w.GetAccounts(wallet)).Returns(new List<HdAccount> { new HdAccount { Name = account } });
+            var consensusManager = new Mock<IConsensusManager>();
 
+            var walletManager = new Mock<IWalletManager>();
+            walletManager.Setup(f => f.GetWalletsNames()).Returns(new List<string> { wallet });
+            walletManager.Setup(f => f.GetAccounts(wallet)).Returns(new List<HdAccount> { new HdAccount { Name = account } });
             HdAddress address = WalletTestsHelpers.CreateAddress();
-            this.walletManager.Setup(w => w.GetUnusedAddress(new WalletAccountReference(wallet, account))).Returns(address);
+            walletManager.Setup(f => f.GetUnusedAddress(new WalletAccountReference(wallet, account))).Returns(address);
 
-            this.powMining.Setup(p => p.GenerateBlocks(It.Is<ReserveScript>(r => r.ReserveFullNodeScript == address.Pubkey), 1, int.MaxValue))
-                .Returns(new List<uint256> { new uint256(1255632623) });
+            var powMining = new Mock<IPowMining>();
+            powMining.Setup(f => f.GenerateBlocks(It.Is<ReserveScript>(r => r.ReserveFullNodeScript == address.Pubkey), 1, int.MaxValue)).Returns(new List<uint256> { new uint256(1255632623) });
+            powNode.Setup(f => f.NodeFeature<MiningFeature>(false)).Returns(new MiningFeature(KnownNetworks.RegTest, new MinerSettings(Configuration.NodeSettings.Default()), Configuration.NodeSettings.Default(), this.loggerFactory, new Mock<ITimeSyncBehaviorState>().Object, powMining.Object, null));
```

Comment:
Further to my other comment, I am not sure why we bother having `this.network` defined if the individual tests in the class operate across different networks.

Reply:
See previous comment ;)